Build Issues with NPM package - Missing icon + no default export

Hi Team, 

Having the following build issues when installing the charts component library with npm: 
 
Error 1:
[node_modules/@rcpch/digital-growth-charts-react-component-library/build/RCPCHChart/RCPCHChart.d.ts(1,8)](mailto:node_modules/@rcpch/digital-growth-charts-react-component-library/build/RCPCHChart/RCPCHChart.d.ts(1,8)): 
error TS1192: Module '"C:/source/Morse/Client/node_modules/@types/react/index"' has no default export.
 
Error 2:
C:\source\Morse\Client\node_modules/@rcpch/digital-growth-charts-react-component-library/build/index.d.ts(2,18):
error TS2307: Cannot find module '../src/images/icon.png'.
 
 
Regarding issue 1, I can resolve by manually editing the ‘RCPCHChart.d.ts’ file and replacing 
import React from 'react';
with
import * as React from 'react'; 
 
For issue 2, the icon doesn't seem to be present in the ‘node_modules\@rcpch\digital-growth-charts-react-component-library’ folder although I can see it in the RCPCH github repo here. Locally resolved by commenting out import and export of icon in ‘index.d.ts’ file. 


I'm using:
"react": "17.0.2",
"typescript": "3.2.4"
"@rcpch/digital-growth-charts-react-component-library": 6.1.4    
"@types/react": "17.0.2"

Any help would be much appreciated!

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
     "name": "@rcpch/digital-growth-charts-react-component-library",
-    "version": "6.1.4",
+    "version": "6.1.5",
     "main": "build/index.js",
     "module": "build/index.esm.js",
     "files": [

--- a/src/CentileChart/CentileChart.stories.tsx
+++ b/src/CentileChart/CentileChart.stories.tsx
@@ -1,5 +1,5 @@
 // Generated with util/create-component.js
-import React from "react";
+import * as React from "react";
 import CentileChart from "./CentileChart";
 
 export default {

--- a/src/CentileChart/CentileChart.test.tsx
+++ b/src/CentileChart/CentileChart.test.tsx
@@ -1,5 +1,5 @@
 // Generated with util/create-component.js
-import React from "react";
+import * as React from "react";
 import { render } from "@testing-library/react";
 
 import CentileChart from "./CentileChart";

--- a/src/CentileChart/CentileChart.tsx
+++ b/src/CentileChart/CentileChart.tsx
@@ -1,4 +1,5 @@
-import React, { useState, useLayoutEffect, useMemo, MouseEvent, useRef } from 'react';
+import * as React from 'react';
+import { useState, useLayoutEffect, useMemo, MouseEvent, useRef } from 'react';
 import {
     // libraries
     createContainer,

--- a/src/RCPCHChart/RCPCHChart.test.tsx
+++ b/src/RCPCHChart/RCPCHChart.test.tsx
@@ -1,5 +1,5 @@
 // Generated with util/create-component.js
-import React from "react";
+import * as React from "react";
 import { render } from "@testing-library/react";
 
 import RCPCHChart from "./RCPCHChart";

--- a/src/RCPCHChart/RCPCHChart.tsx
+++ b/src/RCPCHChart/RCPCHChart.tsx
@@ -1,6 +1,5 @@
 // packages/libraries
-import React from 'react';
-
+import * as React from 'react';
 // props and interfaces
 import { RCPCHChartProps } from './RCPCHChart.types';
 

--- a/src/SDSChart/SDSChart.test.tsx
+++ b/src/SDSChart/SDSChart.test.tsx
@@ -1,5 +1,5 @@
 // Generated with util/create-component.js
-import React from "react";
+import * as React from "react";
 import { render } from "@testing-library/react";
 
 import SDSChart from "./SDSChart";

--- a/src/SDSChart/SDSChart.tsx
+++ b/src/SDSChart/SDSChart.tsx
@@ -1,5 +1,6 @@
 // libraries and frameworks
-import React, { useState, useMemo, MouseEvent, useRef } from "react";
+import * as React from 'react';
+import { useState, useMemo, MouseEvent, useRef } from "react";
 import { 
     VictoryAxis, 
     VictoryChart,

--- a/src/SubComponents/AgeRadioButtonGroup.tsx
+++ b/src/SubComponents/AgeRadioButtonGroup.tsx
@@ -1,4 +1,4 @@
-import React from 'react';
+import * as React from 'react';
 
 export const AgeRadioButtonGroup = (props) => {
     return (

--- a/src/SubComponents/ChartCircle.tsx
+++ b/src/SubComponents/ChartCircle.tsx
@@ -1,4 +1,4 @@
-import React from 'react';
+import * as React from 'react';
 
 export const ChartCircle = (props) =>{ // lollipop tick for months
     const {x, y, style, text} = props

--- a/src/SubComponents/CloseFullScreenIcon.tsx
+++ b/src/SubComponents/CloseFullScreenIcon.tsx
@@ -1,4 +1,4 @@
-import React from 'react';
+import * as React from 'react';
 import styled from 'styled-components';
 import Icon from './Icon'
 

--- a/src/SubComponents/CustomGridComponent.tsx
+++ b/src/SubComponents/CustomGridComponent.tsx
@@ -1,4 +1,4 @@
-import React from 'react';
+import * as React from 'react';
 import { LineSegment } from 'victory';
 
 type PropTypes = {

--- a/src/SubComponents/ErrorBoundary.tsx
+++ b/src/SubComponents/ErrorBoundary.tsx
@@ -1,4 +1,4 @@
-import React from 'react';
+import * as React from 'react';
 import styled from 'styled-components';
 
 import { ChartTitle } from './ChartTitle';

--- a/src/SubComponents/EventCaret.tsx
+++ b/src/SubComponents/EventCaret.tsx
@@ -1,4 +1,4 @@
-import React from 'react';
+import * as React from 'react';
 
 export const EventCaret = (props) => {
     const { x, y, eventsText } = props;

--- a/src/SubComponents/FullScreenIcon.tsx
+++ b/src/SubComponents/FullScreenIcon.tsx
@@ -1,4 +1,4 @@
-import React from 'react';
+import * as React from 'react';
 import styled from 'styled-components';
 import Icon from './Icon'
 

--- a/src/SubComponents/MidParentalHeight.tsx
+++ b/src/SubComponents/MidParentalHeight.tsx
@@ -1,4 +1,4 @@
-import React from 'react';
+import * as React from 'react';
 
 export const MidParentalHeight = (props) => {
 

--- a/src/SubComponents/MonthsLabel.tsx
+++ b/src/SubComponents/MonthsLabel.tsx
@@ -1,4 +1,4 @@
-import React from 'react';
+import *  as React from 'react';
 
 export const MonthsLabel = (props) => { // the same ChartCircle but smaller for use in axis label
     const {x, y, text, style} = props

--- a/src/SubComponents/RenderTickLabel.tsx
+++ b/src/SubComponents/RenderTickLabel.tsx
@@ -1,4 +1,4 @@
-import React from 'react';
+import * as React from 'react';
 
 function RenderTickLabel(props) {
     const x = props.x;

--- a/src/SubComponents/ShareIcon.tsx
+++ b/src/SubComponents/ShareIcon.tsx
@@ -1,4 +1,4 @@
-import React from 'react';
+import * as React from 'react';
 import styled from 'styled-components';
 import Icon from './Icon'
 

--- a/src/SubComponents/XPoint.tsx
+++ b/src/SubComponents/XPoint.tsx
@@ -1,4 +1,4 @@
-import React from 'react';
+import * as React from 'react';
 import { Point } from 'victory';
 
 export const XPoint = (props) => {

--- a/util/templates/component.js
+++ b/util/templates/component.js
@@ -1,6 +1,6 @@
 module.exports = (componentName) => ({
-  content: `// Generated with util/create-component.js
-import React from "react";
+    content: `// Generated with util/create-component.js
+import * as React from "react";
 
 import { ${componentName}Props } from "./${componentName}.types";
 
@@ -13,5 +13,5 @@ const ${componentName}: React.FC<${componentName}Props> = ({ foo }) => (
 export default ${componentName};
 
 `,
-  extension: `.tsx`
+    extension: `.tsx`,
 });

--- a/util/templates/component.stories.js
+++ b/util/templates/component.stories.js
@@ -1,6 +1,6 @@
 module.exports = (componentName) => ({
-  content: `// Generated with util/create-component.js
-import React from "react";
+    content: `// Generated with util/create-component.js
+import * as React from "react";
 import ${componentName} from "./${componentName}";
 
 export default {
@@ -11,5 +11,5 @@ export const WithBar = () => <${componentName} foo="bar" />;
 
 export const WithBaz = () => <${componentName} foo="baz" />;
 `,
-  extension: `.stories.tsx`
+    extension: `.stories.tsx`,
 });

--- a/util/templates/component.test.js
+++ b/util/templates/component.test.js
@@ -1,6 +1,6 @@
 module.exports = (componentName) => ({
-  content: `// Generated with util/create-component.js
-import React from "react";
+    content: `// Generated with util/create-component.js
+import * as React from "react";
 import { render } from "@testing-library/react";
 
 import ${componentName} from "./${componentName}";
@@ -27,5 +27,5 @@ describe("Test Component", () => {
   });
 });
 `,
-  extension: `.test.tsx`
+    extension: `.test.tsx`,
 });

